synapse/lib/remcycle.py

```
@@ -0,0 +1,846 @@
+            self._api_ingests[name].append((action_name, ingest_func, gest_glue))
+
+        self.fire('hypnos:register:api:add', api=name)
+
+    def delConfig(self, namespace):
+        '''
+        Remove a given namespace, APIs and any corresponding event handlers
+        which have been snapped into the Hypnos object and its cortex via
+        the register_config API.
+
+        :param namespace: Namespace of objects to remove.
+        :return: None
+        :raises: NoSuchName if the namespace requested does not exist.
+        '''
+
+        if namespace not in self.namespaces:
+            raise NoSuchName('Namespace is not registered.')
+
+        self.namespaces.remove(namespace)
+        self.docs.pop(namespace, None)
+        self.global_request_headers.pop(namespace, None)
+
+        apis_to_remove = []
+        for api_name in list(self.apis.keys()):
+            ns, name = api_name.split(':', 1)
+            if ns == namespace:
+                apis_to_remove.append(api_name)
+
+        for api_name in apis_to_remove:
+            self._deregister_api(name=api_name)
+
+        self.fire('hypnos:register:namespace:del', namespace=namespace)
+
+    def _deregister_api(self, name):
+        if name not in self.apis:
+            raise NoSuchName('API name not registered.')
+
+        self.apis.pop(name, None)
+
+        funclist = self._api_ingests.pop(name, [])
+        for action_name, ingest_func, gest_glue in funclist:
+            self.off(name, gest_glue)
+            self.core.off(action_name, ingest_func)
+
+        self.fire('hypnos:register:api:del', api=name)
+
+    def getApiNyx(self, name):
+        '''
+        Get the Nyx object corresponding to a given API name.
+
+        :param name: Name of the API to get the object for.
+        :return: A Nyx object.
+        :raises: A NoSuchName error if the requested name does not exist.
+        '''
+        nyx = self.apis.get(name)
+        if not nyx:
```

Done - made the fireApi fail fast on a None object returned, since failing on a duck typing would result in a new job being constructed but never fired, this lingering function references which would eat memory over time.